sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -330,6 +330,17 @@ public void populateDisplayData(DisplayData.Builder builder) {
   }
 
   /**
+   * {@link Object#toString()} wrapper to extract display data values for various types.
+   */
+  private String displayDataString(Object value) {
+    if (value instanceof Object[]) {
```

Comment:
int[] wont be an instanceof Object[]

Might need to use:
https://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#isArray()

Or explicitly do instanceof for all the primitive types.